### ISSUE TITLE
LOCALSTACK_HOSTNAME has wrong value in dockerized localstack

### ISSUE THREAD

**Issue by bmitmanski** [localstack/localstack#2602]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
`LOCALSTACK_HOSTNAME` has wrong value in dockerized localstack
Lambda `LOCALSTACK_HOSTNAME` output: 172.18.0.1

`docker network inspect localstack-network` output: 
```
[
    {
        "Name": "localstack-network",
        "Id": "78fd313ca7b543a63ac8b38463935b78ff436e02c0009b3556d07e09e741d36d",
        "Created": "2020-06-22T12:44:51.398396816Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.18.0.0/16",
                    "Gateway": "172.18.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "71ce4e439e684f7adc18847724ccabc9fa024c5259dda415c6eac2be782409ff": {
                "Name": "ompliance-jetscan-lambdas_pr-187_localstack_1_b6acaac6a401",
                "EndpointID": "488069758fb9911ab1ddfc62ca370d56e032f0e0a06b116fa8ef8b6608f3e185",
                "MacAddress": "02:42:ac:12:00:04",
                "IPv4Address": "172.18.0.4/16",
                "IPv6Address": ""
            },
            ...
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "localstack-network",
            "com.docker.compose.project": "ompliance-jetscan-lambdas_pr-187",
            "com.docker.compose.version": "1.23.1"
        }
    }
]
```

## Expected behavior

`LOCALSTACK_HOSTNAME` has an IP of the localstack container. 

## Actual behavior

`LOCALSTACK_HOSTNAME` has an IP of the Gateway.

# Steps to reproduce

## Command used to start LocalStack

```
localstack:
    image: localstack/localstack
    networks:
      - localstack-network
    ports:
      - 4566:4566
      - 4572:4572
      - 8080:8080
    environment:
      - DEBUG=1
      - HOSTNAME_EXTERNAL=localstack
      - SERVICES=sqs,lambda,cloudwatch,s3,iam,ec2,stepfunctions,cloudwatchlogs
      - LAMBDA_REMOTE_DOCKER=false
      - LAMBDA_EXECUTOR=docker
      - LAMBDA_DOCKER_NETWORK=localstack-network
      - AWS_ACCESS_KEY_ID=123
      - AWS_SECRET_ACCESS_KEY=xyz
      - DEFAULT_REGION=us-east-1
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /tmp/localstack:/tmp/localstack

networks:
  localstack-network:
    name: localstack-network
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
To simplify just print in Lambda:
`console.log(process.env.LOCALSTACK_HOSTNAME)`

**Comment by bmitmanski** [comment]
Workaround here is to just use `localstack` instead of `LOCALSTACK_HOSTNAME`. Docker will resolve this as container name in the network.

**Comment by whummer** [comment]
Thanks for reporting @bmitmanski . This should be fixed in #2633 - the default is now that `LOCALSTACK_HOSTNAME` should point to the container host directly, instead of the Docker bridge IP. Can you please give it another try with the `latest` version? Please report here if the problem persists. Thanks

**Comment by snwinz** [comment]
Hi @whummer,

I'm not sure if this was really a bug.
The documentation says:
`LOCALSTACK_HOSTNAME: Name of the host where LocalStack services are available. Use this hostname as endpoint (e.g., http://${LOCALSTACK_HOSTNAME}:4566) in order to access the services from within your Lambda functions (e.g., to store an item to DynamoDB or S3 from a Lambda).`
Therefore, I would assume, that I can access dynamodb services from within a lambda in a container for example with `const dynamodb = new AWS.DynamoDB({endpoint: `http://${process.env.LOCALSTACK_HOSTNAME}:4566`});`
However, this does not work since it is resolved to **172.18.0.2** when I'm using docker-compose. But it works if I use the ip of the gateway **172.18.0.1** which seems to be the original value of LOCALSTACK_HOSTNAME.
